Runtime/Scripts/PersistentConnection.cs

```diff
@@ -131,192 +217,193 @@ IEnumerator OnIntervalEdgeEventsLocation(EdgeEventsConnection connection)
         if (locationUpdatesCounter >= config.locationConfig.maxNumberOfUpdates)
         {
           locationUpdatesRunning = false;
+          Logger.Log("Stopping Location Updates according to configs, No. LocationUpdates: " + locationUpdatesCounter);
+          locationUpdatesCounter = 0;
           yield break;
         }
       }
       yield return new WaitForSecondsRealtime(config.locationConfig.updateIntervalSeconds);
       yield return StartCoroutine(LocationService.EnsureLocation());
       Loc location = LocationService.RetrieveLocation();
+      Logger.Log("EdgeEvents Posting location update");
       connection.PostLocationUpdate(location);
       locationUpdatesCounter++;
       yield return StartCoroutine(OnIntervalEdgeEventsLocation(connection));
     }
 
     IEnumerator OnIntervalEdgeEventsLatency(EdgeEventsConnection connection, string host)
+    {
+      if (config.latencyConfig.maxNumberOfUpdates > 0)
+      {
+        if (latencyUpdatesCounter >= config.latencyConfig.maxNumberOfUpdates)
         {
-          if (config.latencyConfig.maxNumberOfUpdates > 0)
-          {
-            if (latencyUpdatesCounter >= config.latencyConfig.maxNumberOfUpdates)
+          latencyUpdatesRunning = false;
+          Logger.Log("Stopping Latency Updates according to configs, No. LatencyUpdates: " + latencyUpdatesCounter);
+          latencyUpdatesCounter = 0;
+          yield break;
+        }
+      }
+      yield return new WaitForSecondsRealtime(config.latencyConfig.updateIntervalSeconds);
+      yield return StartCoroutine(LocationService.EnsureLocation());
+      if (hasTCPPorts)
+      {
+        connection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
+      }
+      else
+      {
+        integration.matchingEngine.EdgeEventsConnection
+          .TestPingAndPostLatencyUpdate
+          (integration.GetHost(), location).ConfigureAwait(false);
```

Comment:
Can this just be `connection.TestPingAndPostLatencyUpdate` like it is on line 250?